### ISSUE TITLE
MD044 should ignore strings in http links

### ISSUE THREAD

**Issue by alexbuckgit** [DavidAnson/markdownlint#358]
MD044 should be configurable to ignore lowercasing in http links, similar to the code_blocks property.

**Comment by DavidAnson** [comment]
Note to self: The request is to add a new property that tells this rule to ignore case mismatches in URLs so that a proper name can be forced to be correctly-cased everywhere in the document but not in a case-sensitive URL.

**Comment by DavidAnson** [comment]
I think I misunderstood the request. The following Markdown reports only a single violation, ignoring the "wrong" case in the URL. Could you please give an example of something that does not work as you expect?

```markdown
# Issue 358

The proper case is: JavaScript

So this is wrong: javascript

But it's okay in a URL: https://example.org/javascript.txt

<!-- markdownlint-configure-file {
    "proper-names": {
        "names": [
            "JavaScript"
        ]
    },
    "no-bare-urls": false
} -->
```

Interactive demo:
https://dlaa.me/markdownlint/#%25m%23%20Issue%20358%0A%0AThe%20proper%20case%20is%3A%20JavaScript%0A%0ASo%20this%20is%20wrong%3A%20javascript%0A%0ABut%20it's%20okay%20in%20a%20URL%3A%20https%3A%2F%2Fexample.org%2Fjavascript.txt%0A%0A%3C!--%20markdownlint-configure-file%20%7B%0A%20%20%20%20%22proper-names%22%3A%20%7B%0A%20%20%20%20%20%20%20%20%22names%22%3A%20%5B%0A%20%20%20%20%20%20%20%20%20%20%20%20%22JavaScript%22%0A%20%20%20%20%20%20%20%20%5D%0A%20%20%20%20%7D%2C%0A%20%20%20%20%22no-bare-urls%22%3A%20false%0A%7D%20--%3E%0A
